bezier_library/src/bezier_grinding_surfacing.cpp

```diff
@@ -758,92 +756,98 @@ bool BezierGrindingSurfacing::filterNeighborPosesTooClose(BezierPointNormalTable
+                                   const Eigen::Vector3d &first_point_normal,
+                                   const Eigen::Vector3d &last_point,
+                                   const Eigen::Vector3d &last_point_normal,
+                                   const Eigen::Vector4d &plane_equation,
+                                   EigenSTL::vector_Affine3d &trajectory)
 {
-  // The extrication filter compare the angle between a point of the grinding line and all the extrication
-  // points located in the side where the grinding point is placed.
-  // The filter works in two parts :
-  // - The first part use as a reference point the last point of the grinding line N. It compute the angle between
-  // the normal of this point and the vector between the reference point and the extrication point.
-  // as long as the value of the angle is outside the bounds defined by upper_tolerance and lower_tolerance,
-  // the extrication point is filtered. When the angle fall inside the bounds, the process for this side is stopped.
-  // - The same process is repeated on the other side of the extrication line, using as reference point, the first point
-  // of the grinding line N + 1
-  // At the end of theses process, all the point which haven't been filtered are removed from the trajectory vector
-
-  if (trajectory.empty())
+  // The extrication filter use a 2D angle comparison method to reject inaccessible points from the extrication trajectory.
+  // The filter apply the comparison algorithm on both side of the extrication trajectory :
+  // - At the beginning of the trajectory, it takes the normal (Nor) of the last grinding point of the current grinding line (GN)
+  // - and make a projection of this vector onto the plane used to slice the mesh and generate the extrication trajectory.
+  // - It then computes a vector (AN) between the last grinding point A a extrication point N. Thanks to the projection made before,
+  // - the normal (Nor) and the vector (AN) are located into the same 2D plane. An angle computation it's performed.
+  // - The sign of the first angle computed is taken as reference. The operation is then repeated with the
+  // - following points (N+1, N+2 ...) as long as the current angle computed has the same sign than the reference angle.
+  // - A change of sign occur when we reach a point (T) located just above the grinding point, thus
+  // - accessible for the robot. Finally all the point located before (T) are removed from the trajectory.
+  // - The same process is repeated at the end side of the trajectory, but the normal (Nor) is the normal at the first point of the
+  // - next grinding line (GN +1) trajectory.
+  // The formula used to project the vector is : A - (A.n)n with A the vector to project and N the normal of the plane
+  // The formula used to compute the 2D angle is : arctan(determinant, (dot product))
+
+  if (trajectory.size() <= 4)
     return false;
```

Comment:
Do we really need at least 5 ponts ? I think it works with 2 points.

Maybe add a `ROS_ERROR_STREAM("BezierGrindingSurfacing::filterExtricationTrajectory: Trajectory is too small!")`